cpp/src/parquet/encoding-test.cc

```diff
@@ -314,6 +318,185 @@ TEST(TestDictionaryEncoding, CannotDictDecodeBoolean) {
+}
+
+TEST_F(TestDecodeArrowPlain, DecodeNonNullDictionary) {
+  ::arrow::BinaryDictionaryBuilder builder(default_memory_pool());
+  auto actual_num_values = decoder_->DecodeArrowNonNull(num_values_, &builder);
+  CheckDecodeDict(actual_num_values, builder);
+}
+
+// ----------------------------------------------------------------------
+// Arrow builder decode tests for DictByteArrayDecoder
+class TestDecodeArrowDict : public TestDecodeArrow {
+ public:
+  void SetupEncoderDecoder() override {
+    auto node = schema::ByteArray("name");
+    descr_ = std::unique_ptr<ColumnDescriptor>(new ColumnDescriptor(node, 0, 0));
+    encoder_ = MakeTypedEncoder<ByteArrayType>(Encoding::PLAIN, /*use_dictionary=*/true,
+                                               descr_.get());
+    ASSERT_NO_THROW(encoder_->Put(input_data_.data(), num_values_));
+    buffer_ = encoder_->FlushValues();
+
+    auto dict_encoder = dynamic_cast<DictEncoder<ByteArrayType>*>(encoder_.get());
+    ASSERT_NE(dict_encoder, nullptr);
+    dict_buffer_ =
+        AllocateBuffer(default_memory_pool(), dict_encoder->dict_encoded_size());
+    dict_encoder->WriteDict(dict_buffer_->mutable_data());
+
+    // Simulate reading the dictionary page followed by a data page
+    decoder_ = MakeTypedDecoder<ByteArrayType>(Encoding::PLAIN, descr_.get());
+    decoder_->SetData(dict_encoder->num_entries(), dict_buffer_->data(),
+                      static_cast<int>(dict_buffer_->size()));
+
+    dict_decoder_ = MakeDictDecoder<ByteArrayType>(descr_.get());
+    dict_decoder_->SetDict(decoder_.get());
+    dict_decoder_->SetData(num_values_, buffer_->data(),
+                           static_cast<int>(buffer_->size()));
+  }
+
+  void TearDown() override {}
+
+ protected:
+  std::unique_ptr<ColumnDescriptor> descr_;
+  std::unique_ptr<DictDecoder<ByteArrayType>> dict_decoder_;
+  std::shared_ptr<Buffer> dict_buffer_;
+};
+
+TEST_F(TestDecodeArrowDict, DecodeDense) {
+  ::arrow::internal::ChunkedBinaryBuilder builder(static_cast<int>(dict_buffer_->size()),
+                                                  default_memory_pool());
+  auto byte_array_decoder = dynamic_cast<ByteArrayDecoder*>(dict_decoder_.get());
+  ASSERT_NE(byte_array_decoder, nullptr);
+  auto actual_num_values =
+      byte_array_decoder->DecodeArrow(num_values_, 0, valid_bits_.data(), 0, &builder);
+  CheckDecodeDense(actual_num_values, builder);
+}
+
+TEST_F(TestDecodeArrowDict, DecodeNonNullDense) {
+  ::arrow::internal::ChunkedBinaryBuilder builder(static_cast<int>(dict_buffer_->size()),
+                                                  default_memory_pool());
+  auto byte_array_decoder = dynamic_cast<ByteArrayDecoder*>(dict_decoder_.get());
+  ASSERT_NE(byte_array_decoder, nullptr);
+  auto actual_num_values = byte_array_decoder->DecodeArrowNonNull(num_values_, &builder);
+  CheckDecodeDense(actual_num_values, builder);
+}
+
+TEST_F(TestDecodeArrowDict, DecodeDictionary) {
```

Comment:
For consistency, consider either naming this `DecodeDict`, or spelling out `Dictionary` in full elsewhere in this diff.